roles/foreman_proxy_content/tasks/install.yml

```
@@ -1,24 +1,12 @@
 ---
-- name: 'Get Server Hostname'
-  shell: hostname -f
-  delegate_to: "{{ foreman_proxy_content_server }}"
-  register: server_fqdn
-
-- name: 'Get Server IP Address'
-  shell: ifconfig eth0 2>/dev/null|awk '/inet/ {print $2}'|sed 's/addr://'
+- name: 'Gather facts from {{ foreman_proxy_content_server }}'
+  setup:
   delegate_to: "{{ foreman_proxy_content_server }}"
-  register: server_ip
+  delegate_facts: true
 
-- name: 'Set fact for Server IP'
-  set_fact: server_ip="{{ server_ip.stdout.split('\n')[0] }}"
-
-- name: 'Add /etc/hosts record for foreman_proxy_content'
-  delegate_to: "{{ foreman_proxy_content_server }}"
-  lineinfile:
-    dest: /etc/hosts
-    regexp: ".*{{ ansible_nodename }}$"
-    line: "{{ foreman_proxy_content_ip }} {{ ansible_nodename }}"
-    state: present
+- name: 'Get Server Hostname'
+  set_fact:
+    server_fqdn: "{{ hostvars[foreman_proxy_content_server].ansible_nodename }}"
```

Perhaps we can classify this as well?
```suggestion
    foreman_server_fqdn: "{{ hostvars[foreman_proxy_content_server].ansible_nodename }}"
```

Perhaps even expand it (no idea if it's valid, but you get the idea):
```suggestion
    foreman_server_fqdn: "{{ hostvars[foreman_proxy_content_server].ansible_nodename }}"
    foreman_server_url: "https://{{ foreman_server_fqdn }}"
```

Alternatively we could read the config. `foreman-rake config` has a few settings we need (`foreman_url`, `oauth_consumer_key` and `oauth_consumer_secret`). I don't think it currently can easily output JSON, but that would be ideal. Perhaps a good enhancement.